### PR TITLE
Add IVA support to Claims Repository (GSI-649)

### PR DESCRIPTION
**Pull request by Cito**
This PR adds IVA support to the claims repository, as mentioned in the [epic spec](https://github.com/ghga-de/epic-docs/blob/main/46-sugar-ant/technical_specification.md#claims-repository).

### CODE REVIEW

File: tests/integration/user_management/claims_repository/test_api.py [hunk 1]
```diff
@@ -333,17 +351,102 @@ async def test_grant_download_access(client_with_db):
...
+    assert claim_data["user_id"] == "john@ghga.de"
+    assert claim_data["iva_id"] == "some-iva-id"
+
+    response = await client_with_db.get("/download-access/users/john@ghga.de/datasets")
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json() == []
+
+
+async def test_grant_download_access_without_iva(client_with_db):
+    """Test that granting access to a dataset when the IVA does not exist."""
```
Comment (TheByronHimes) [inline tests/integration/user_management/claims_repository/test_api.py:396]:
```suggestion
    """Test granting access to a dataset when the IVA does not exist."""
```

File: tests/integration/user_management/claims_repository/test_api.py [hunk 2]
```diff
@@ -333,17 +351,102 @@ async def test_grant_download_access(client_with_db):
     assert response.json() == ["some-dataset-id"]
 
 
+async def test_grant_download_access_with_unverified_iva(client_with_db):
+    """Test that granting access to a dataset when the IVA is not yet verified."""
```
Comment (TheByronHimes) [inline tests/integration/user_management/claims_repository/test_api.py:355]:
```suggestion
    """Test granting access to a dataset when the IVA is not yet verified."""
```

**Review by TheByronHimes — DISMISSED** [review]
Looks good as far as I can see.

File: src/auth_service/user_management/claims_repository/models/claims.py [hunk 3]
```diff
@@ -190,6 +194,9 @@ class ClaimFullCreation(ClaimCreation):
     user_id: str = Field(  # actually UUID
         default=..., description="Internal user ID"
     )
+    iva_id: Optional[str] = Field(  # actually UUID
+        default=None, description="ID of the IVA associated with this claim"
+    )
```
[the remark this answers is not in the document]
Comment (Cito) [reply src/auth_service/user_management/claims_repository/models/claims.py]: No, you're right, it's superfluous here. Fixed in 0c15d5d6.

**Review by TheByronHimes — APPROVED** [review]

**Pull request merged**